cpa/_load.py

```diff
@@ -0,0 +1,36 @@
+"""Functionalities to load data from splits to CPA AnnData with the correct fields."""
+
+import scanpy as sc
+from pathlib import Path
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def _validate_cols(adata: sc.AnnData) -> bool:
+    """Validate that CPA-required columns are present."""
+    required_cols = ['split', 'condition', 'dose_val']
+    for col in required_cols:
+        if col not in adata.obs.columns:
+            logging.error(f'Column {col} not found in adata.obs')
+            return False
+    return True
+
+def load_split(train_path: str,
+               valid_path: str,
+               test_path: str,
+               control_condition: str = 'GFP') -> sc.AnnData:
+    """Load a train/valid/test split and format for CPA."""
+    adata_train = sc.read_h5ad(train_path)
+    adata_valid = sc.read_h5ad(valid_path)
+    adata_test = sc.read_h5ad(test_path)
+    adata = adata_train.concatenate([adata_valid, adata_test],
+                                      batch_key='split',
+                                        batch_categories=['train', 'test', 'ood'])
+    assert _validate_cols(adata), "Required columns not found in adata.obs"
```

Comment:
May be overkill, but it could be nice to check each of the train, valid, and test datasets for the required columns individually, so that if the test fails, you know which ones are missing those columns.  

Also, I believe the `concatenate` operation will do an inner join on the column variables automatically, so that if any one of the three datasets are missing a required column, the concatenated dataset will also be, and will fail the test (desired behavior).  But checking each dataset individually will guard against a possible change to this function (say, in a later version of the AnnData package) that makes an outer join is the default behavior, such that even if one or two of the datasets are missing a required column, it will show up in the concatenated dataset as long as one of the three has it.  Again, perhaps paranoid.  🙃

Reply:
I agree! And I think it makes sense to explicitly throw an error so you can tell the user exactly what happened rather than a general assertion error.

Reply:
Makes sense to me too!